rust/sbp/src/de.rs

```diff
@@ -174,17 +126,64 @@ impl std::fmt::Display for CrcError {
 
 impl std::error::Error for CrcError {}
 
-#[derive(Debug, Clone, Default)]
-struct SbpDecoder;
+struct Framer<R>(FramedRead<R, FramerImpl>);
```

Comment:
pub struct here (also you should `pub use` it from lib.rs (same w decoder))

Reply:
do we want decoder to be public? its just constructed in iter_messages()

Reply:
I'd like to deprecate `iter_messages`, in hindsight I think exposing a named type is better than the `-> impl Iterator`. You can't store `impl Iterator` in a field of a struct without `Box`. We could also add more helper methods to `Decoder`. And it's not any less convenient to use (it's shorter actually)
```
for m in sbp::Decoder::new(reader) {}
vs
for m in sbp::iter_messages(reader) {}
```